CIScripts/Tests/TCPCommunicationTest.ps1

```
@@ -1,3 +1,67 @@
 function Test-TCPCommunication {
-    # TODO: JW-838: Implement
+    Param ([Parameter(Mandatory = $true)] [System.Management.Automation.Runspaces.PSSession] $Session,
+           [Parameter(Mandatory = $true)] [TestConfiguration] $TestConfiguration)
+
+    Write-Host "Running TCP Communication test"
+
+    Initialize-TestConfiguration -Session $Session -TestConfiguration $TestConfiguration
+
+    $NetworkName = $TestConfiguration.DockerDriverConfiguration.NetworkConfiguration.NetworkName
+
+    Write-Host "Creating containers"
+    $ServerID, $ClientID = Invoke-Command -Session $Session -ScriptBlock {
+        $ServerID = docker run --network $Using:NetworkName -d iis-tcptest
+        $ClientID = docker run --network $Using:NetworkName -d microsoft/nanoserver ping -t localhost
+        return $ServerID, $ClientID
+    }
+
+    . $PSScriptRoot\CommonTestCode.ps1
+
+    Write-Host "Getting VM NetAdapter Information"
+    $VMNetInfo = Get-RemoteNetAdapterInformation -Session $Session -AdapterName $TestConfiguration.AdapterName
+
+    Write-Host "Getting Containers NetAdapter Information"
+    $ServerNetInfo = Get-RemoteContainerNetAdapterInformation -Session $Session -ContainerID $ServerID
+    $ClientNetInfo = Get-RemoteContainerNetAdapterInformation -Session $Session -ContainerID $ClientID
+
+    Write-Host "Configuring..."
+    Invoke-Command -Session $Session -ScriptBlock {
```

Would it be possible to add some error checking here? For example, by analyzing error codes from utilities.

---

Failure of any utility command will print error to stdout. It is true that script will not stop on that error and will try to execute all further commands but it will fail finally on `Invoke-WebRequest`, so I think it is more readable to ignore all errors there.